tests/functional/dcp-prototype/backend/browser/test_api.py

```diff
@@ -3,6 +3,10 @@
 import unittest
 
 import requests
+from dcplib.aws_secret import AwsSecret
+
+if not os.getenv("DEPLOYMENT_STAGE"):  # noqa
+    os.environ["DEPLOYMENT_STAGE"] = "test"  # noqa
```

Comment:
Is it possible to use `DEPLOYMENT_STAGE=dev` to test auth? The test deployment is planned to be deprecated

Reply:
isn't `test` going to be used for local tests?

Reply:
This should all still work as expected once the deployed test infra is destroyed and we change the test domain to localhost.

Reply:
The test infra also includes the database, s3 bucket, secret which is controlled by `DEPLOYMENT_STAGE`.

These tests run against deployed environments for the following reasons:
1. Functional tests would run following a deploy
2. We (previously) were not accessing AWS credentials in push tests

This PR resolves (2) and (1) is a decision we can re-visit. It would be great if functional tests ran against a local deployment during push tests on PRs. However, this would require the test infra to remain, specifically to preserve the S3 bucket and database. For simplicity, I'd propose the entire test infra remains, in which case the deployed test API would not be used in functional tests and instead used as a dummy environment during development. Thoughts?

Reply:
Actually, the test infra requirements boil down to the DB credentials in Secrets Manager so that could be pretty simple. Pinged you to chat over VC